Documentation/gettingstarted/kubeproxy-free.rst

```diff
@@ -1245,16 +1245,48 @@ working, take a look at `this KEP
 Neighbor Discovery
 ******************
 
-When kube-proxy replacement is enabled Cilium does L2 neighbor discovery of nodes in the cluster.
-In some rare cases Cilium may leave stale entries behind in the neighbor table causing packets
-between some nodes to be dropped. To prevent Cilium from performing the neighbor discovery and
-instead rely on the Linux kernel to discover hosts on the same L2 network you can pass the
-``--enable-l2-neigh-discovery=false`` flag to the cilium-agent. However note that relying on the
-Linux Kernel might also cause some packets to be dropped, e.g., a NodePort request can be dropped on
-an intermediate node (i.e., the one which received and is going to forward to a destination node
-which runs the selected service endpoint) if there is no L2 neigh entry in the kernel (due to the
-entry being garbage collected or that ARP resolution has not been done by the kernel). This is
-because Cilium does not drive the ARP resolution from the BPF programs.
+When kube-proxy replacement is enabled, Cilium does L2 neighbor discovery of nodes
+in the cluster. This is required for the service load-balancing in order to populate
+L2 addresses for backends for the fast-path where it is not possible to dynamically
+resolve neighbors on demand.
+
+In Cilium 1.10 or earlier, the agent itself contained an ARP resolution library
+where it triggered discovery and periodic refresh of new nodes joining the cluster.
+The resolved neighbor entries were pushed into the kernel and refreshed as PERMANENT
+entries. In some rare cases, Cilium 1.10 or earlier might have left stale entries behind
+in the neighbor table causing packets between some nodes to be dropped. To prevent from
+performing the neighbor discovery and instead relying on the Linux kernel to discover
+hosts on the same L2 network you can pass the ``--enable-l2-neigh-discovery=false``
+flag to the cilium-agent. However, note that relying on the Linux Kernel might also
+cause some packets to be dropped, for example, a NodePort request can be dropped on
+an intermediate node (i.e., the one which received a service packet and is going to
+forward it to a destination node which runs the selected service endpoint). This
+could happen if there is no L2 neighbor entry in the kernel (due to the entry being
+garbage collected or given that the neighbor resolution has not been done by the kernel).
+This is because it is not possible to drive the neighbor resolution from BPF programs
+in the fast-path e.g. at the XDP layer.
```

Comment:
Overall, this new documentation section feels like exactly what we'd put in the release blog post. It's a bit unusual I think to have the documentation for a given version (v1.11) describe the behavior of previous versions.

Reply:
I think it may be useful given most folks are still on v1.10 or earlier and these are two different approaches, so might be good to know what changed in terms of neighbor discovery. We can remove the before/after in 1.12, but I think it'd be good to have this info here.